src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -69,8 +69,18 @@ function (indicator_hg::IndicatorHennemannGassner)(u, mesh::Union{TreeMesh{1}, S
     end
 
     # Calculate energy in higher modes
-    energy = max((total_energy - total_energy_clip1) / total_energy,
-                 (total_energy_clip1 - total_energy_clip2) / total_energy_clip1)
+    tolE = eps(total_energy)
+    if total_energy > tolE
+      energy_frac_1 = (total_energy - total_energy_clip1) / total_energy
+    else
+      energy_frac_1 = 0.0
+    end
+    if total_energy_clip1 > tolE
+      energy_frac_2 = (total_energy_clip1 - total_energy_clip2) / total_energy_clip1
+    else
+      energy_frac_2 = 0.0
+    end
+    energy = max(energy_frac_1, energy_frac_2)
```

Comment:
I would prefer the following solution:
```suggestion
    if iszero(total_energy)
      energy_frac_1 = (total_energy - total_energy_clip1) / total_energy
    else
      energy_frac_1 = zero(total_energy)
    end
    if iszero(total_energy_clip1)
      energy_frac_2 = (total_energy_clip1 - total_energy_clip2) / total_energy_clip1
    else
      energy_frac_2 = zero(total_energy_clip1)
    end
    energy = max(energy_frac_1, energy_frac_2)
```

This is faster than calculating `eps(total_energy)` all the time and type stable.